File: project-euler/010/problem010.py
#!/usr/bin/env python

from itertools import takewhile


def primes():
    D = {}
    q = 2

    while True:
        p = D.pop(q, None)
        if p:
            x = p + q
            while x in D:
                x += p
            D[x] = p
        else:
            D[q * q] = q
            yield q
        q += 1


print sum(takewhile(lambda x: x < 2000000, primes()))

File: project-euler/010/problem010.rb
#!/usr/bin/env ruby
class Integer
  def prime?
    return false unless self > 1
    (2..Math.sqrt(self)).any? do |x|
      next if x == self
      return false if (self % x).zero?
    end
    true
  end
end

puts (0..2_000_000).select(&:prime?).reduce(:+)
